ligo/main.mligo

```diff
@@ -127,8 +132,21 @@ let set_position (s : storage) (i_l : tick_index) (i_u : tick_index) (i_l_l : ti
         else
             ticks) in
     (* delete the position if liquidity has fallen to 0 *)
-    let position_entry : position_state option = if liquidity_new = 0n then None else Some {position with liquidity = liquidity_new} in
-    let positions = Big_map.update position_key position_entry s.positions in
+    let (positions, position_indexes, new_position_id) =
+        if liquidity_new = 0n then
+            ( Big_map.update position_key (None : position_state option) s.positions
+            , Big_map.update position.position_id (None : position_index option) s.position_indexes
```

Comment:
Minor: there are `Map.remove` and `Map.add` methods, with them this could be written shorter.

Reply:
* `Big_map.remove` and `Big_map.add`

Reply:
* `Big_map.remove` and `Big_map.add`

Reply:
Updated this.